src/LaunchGrant.sol

```diff
@@ -6,15 +6,19 @@ import { IGrant } from './IGrant.sol';
 contract LaunchGrant is IGrant {
     uint256 internal immutable launchDayTimestampInSeconds  = 1690167600; // Monday, 24 July 2023 03:00:00
 
-    function getCurrentId() external view override returns (uint256) {
-        uint weeksSinceLaunch = (block.timestamp - launchDayTimestampInSeconds) / 1 weeks;
-        // Monday, 24 July 2023 03:00:00 until Monday, 07 August 2023 02:59:59 (2 weeks)
+    function calculateId(uint256 timestamp) external pure returns (uint256) {
+        uint weeksSinceLaunch = (timestamp - launchDayTimestampInSeconds) / 1 weeks;
```

Comment:
Potential underflow

Reply:
Should revert if negative. No valid grant before launch. Need to start at when we hand out reservations.